core/src/main/java/org/bitcoinj/core/TransactionOutPoint.java

```diff
@@ -70,21 +70,46 @@ public class TransactionOutPoint {
     public static TransactionOutPoint read(ByteBuffer payload) throws BufferUnderflowException, ProtocolException {
         Sha256Hash hash = Sha256Hash.read(payload);
         long index = ByteUtils.readUint32(payload);
-        return new TransactionOutPoint(index, hash);
+        return TransactionOutPoint.of(hash, index);
     }
 
+    /**
+     * @deprecated Use {@link TransactionOutPoint#from(Transaction, long)}
+     */
+    @Deprecated
     public TransactionOutPoint(long index, Transaction fromTx) {
         this(fromTx.getTxId(), index, fromTx, null);
     }
 
+    @Deprecated
     public TransactionOutPoint(long index, Sha256Hash hash) {
         this(hash, index, null, null);
     }
 
+    private TransactionOutPoint(Sha256Hash hash, long index) {
+        this(hash, index, null, null);
+    }
+
+    /**
+     * @deprecated Use {@link TransactionOutPoint#from(TransactionOutput)}
+     */
+    @Deprecated
     public TransactionOutPoint(TransactionOutput connectedOutput) {
         this(connectedOutput.getParentTransactionHash(), connectedOutput.getIndex(), null, connectedOutput);
     }
 
+    public static TransactionOutPoint of(Sha256Hash hash, long index) {
```

Comment:
This needs a JavaDoc. If this constructor is for unconnected outpoints (as you say in the commit description), I think we could pick a less generic method name, e.g. `unconnected()`?

Reply:
JavaDoc added and force-pushed.

Since a `TransactionOutPoint` is defined as (TxID, index) I think the generic method name `of()` is the best choice. I'd be more open to renaming the `from()` static constructors to `connected()`, but I think `from()` is a better choice as it is simple and IMO more clear about what is happening. (see the added JavaDoc.)